users/views.py

```diff
@@ -1,3 +1,81 @@
-from django.shortcuts import render
+from django.contrib.auth import get_user_model
+from django.core.mail import send_mail
+from django.conf import settings
+from django.urls import reverse
+from rest_framework.response import Response
+from rest_framework import status, generics
+from rest_framework.permissions import AllowAny
+from rest_framework.views import APIView
+from .utils import generate_verification_token, verify_token
+from .serializers import UserSerializer 
+
+User = get_user_model()
+
+
+class RegisterView(generics.CreateAPIView):
+    """
+    View for registering a new user.
+
+    This view handles the creation of a new user and sends a verification email
+    to confirm their email address.
+    """
+
+    permission_classes = [AllowAny]
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+    def perform_create(self, serializer):
+        """
+        Save the new user instance and send a verification email.
+
+        Args:
+            serializer: The serializer instance containing the user data.
+
+        Returns:
+            None
+        """
+        user = serializer.save(is_active=False)
+
+        token = generate_verification_token(user)
+        verification_url = f"http://127.0.0.1:8000/auth/verify-email/?token={token}"
```

Comment:
Hardcoding `http://127.0.0.1:8000` for the verification URL is not ideal. Instead, use `settings.FRONTEND_URL` (or another configurable setting) to dynamically generate the verification link.
```
verification_url = f"{settings.FRONTEND_URL}/auth/verify-email/?token={token}"
```
Then, in `settings.py`, configure:
```
FRONTEND_URL = "https://your-frontend-url.com"
```
This ensures it works in different environments.